### PR TITLE
Remove Pronto and setup Danger

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened

Remove  Pronto and setup Danger for automated code review

## Insight

- Remove completely `pronto` related things
- Add `danger` gem and its plugins
- Create a new workflow for danger checks:
  + I want danger to run as fast as possible, so I create a new workflow and just install node, ruby, bundler and run dependencies installers instead of using docker
  + Add caching to improve the checks
- Add `team-nimblehq` as bot:
  + Add `team-nimblehq` to repo with `write` access (use `write` instead of `read` as we need to set the status of the PR)
  + Create a new token for `team-nimblehq` with `public_repo` access
- Because of removing `pronto-eslint_npm` that includes `scss_lint`, we have to install `scss_lint` to use scss_lint
- List of danger plugins:
```ruby
  gem 'danger-brakeman_scanner' # Security static analysis scanner in danger
  gem 'danger-eslint' # Lint JavaScript code
  gem 'danger-rails_best_practices' # Analyze code regarding best practices
  gem 'danger-reek' # Detect code smell
  gem 'danger-rubocop' # A Danger plugin for Rubocop
  gem 'danger-slim_lint' # Lint slim files
  gem 'danger-suggester' # Suggest code changes based on configured code
```
- Remove duplicated checks in `codebase_spec`
- Run workflow with `pull_request` event and `[opened, reopened, synchronize]` types

## Proof Of Work

 We should have automated reviews on this PR

### CODE REVIEW

**Comment by team-nimblehq** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: undefined local variable or method `rails_best_practices` for #\<Danger::Dangerfile:0x0000563efcc5b7f0\>
```
 #  from Dangerfile:18
 #  -------------------------------------------
 #  # Runs rails_best_practices on modified and added files in the PR
 >  rails_best_practices.lint
 #  
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by team-nimblehq** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Finish test</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Gemfile [hunk 1]
```diff
@@ -50,15 +50,18 @@ group :development, :test do
   gem 'brakeman', require: false # A static analysis security vulnerability scanner for Ruby on Rails applications
```
Comment (team-nimblehq) [inline Gemfile:50]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/e2e4612e1a042effe4fb416e743d1c929888c14f/Gemfile#L50"></span>Gems should be sorted in an alphabetical order within their section of the Gemfile. Gem <code>brakeman</code> should appear before <code>sassc-rails</code>.</td>
    </tr>
  </tbody>
</table>

File: Gemfile.lock [hunk 2]
```diff
@@ -436,6 +457,16 @@ DEPENDENCIES
...
   bullet
   capybara (>= 2.15)
+  danger
+  danger-brakeman_scanner
+  danger-eslint
+  danger-lgtm
+  danger-rails_best_practices
+  danger-reek
+  danger-rubocop
+  danger-simplecov_json
```
Comment (team-nimblehq) [inline Gemfile.lock:467]:
```suggestion

```



<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/javascript/packs/application.js [hunk 3]
```diff
@@ -12,6 +12,8 @@ import 'translations/translations';
 import 'initializers/';
 import 'screens/';
 
+let hoang = "test"
```
Comment (team-nimblehq) [inline app/javascript/packs/application.js:15]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/9026022001e8e25853aba46c94d7bfb1f3d5d7bf/app/javascript/packs/application.js#L15"></span>‘hoang’ is assigned a value but never used.</td>
    </tr>
  </tbody>
</table>

Comment (team-nimblehq) [inline app/javascript/packs/application.js:15]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/9026022001e8e25853aba46c94d7bfb1f3d5d7bf/app/javascript/packs/application.js#L15"></span>Strings must use singlequote.</td>
    </tr>
  </tbody>
</table>

Comment (team-nimblehq) [inline app/javascript/packs/application.js:15]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/9026022001e8e25853aba46c94d7bfb1f3d5d7bf/app/javascript/packs/application.js#L15"></span>Missing semicolon.</td>
    </tr>
  </tbody>
</table>

**Comment by team-nimblehq** [comment]
<p align='center'>LGTM</p>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Dangerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Warns when there is a big PR
+warn('Big PR') if git.lines_of_code > 500
```
Comment (olivierobert) [inline Dangerfile:4]: Where did you get the 500 lines of code number from? Not sure if it's a good number or not 😓 
As an improvement, how about being more detailed in the message e.g "This PR is too large. Prefer splitting it into smaller PRs." 💭

File: Dangerfile [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (olivierobert) [inline Dangerfile:5]: All checks below now duplicate what we have in `spec/codebase/codebase_spec.rb`. So we should revisit `codebase_spec.rb` to see what we need to keep and remove.

File: Dangerfile (hunk 4 above)
Comment (hoangmirs) [reply Dangerfile:4]: This is the default command, let me remove it, we can comment directly in PR

File: Dangerfile (hunk 5 above)
Comment (hoangmirs) [reply Dangerfile]: We need to discuss more about this.
The `spec/codebase/codebase_spec.rb` is used to check on the test, we can use it on local. But danger will run on CI, if we remove the duplicated tests out of codebase, we have to run the checks manually.

**Pull request merged**